src/Time/Time.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Time;
+
+use Zest\Common\TimeZone;
+use Zest\Contracts\Time\Time as TimeContract;
+
+class Time implements TimeContract
+{
+    /**
+     * Get "now" time.
+     *
+     * @param string $timezone Valid php supported timezone.
+     *
+     * @since 3.0.0
+     *
+     * @return int
+     */
+    public static function now($timezone = '')
+    {
+        TimeZone::seteDefaultTz($timezone);
+
+        return time();
+    }
+
+    /**
+     * Number of days in a month.
+     *
+     * @param int $month A numeric month.
+     * @param int $year  A numeric year.
+     *
+     * @since 3.0.0
+     *
+     * @return int
+     */
+    public static function daysInMonth($month = 1, $year = 1970)
+    {
+        return cal_days_in_month(CAL_GREGORIAN, $month, $year);
+    }
+
+    /**
+     * Determine whether the year is leap or not.
+     *
+     * @param int $year A numeric year.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+    public static function isLeapYear($year = null)
+    {
+        $year = $year ?? date('Y');
+
+        return (($year % 4 == 0) && ($year % 100 != 0)) || ($year % 400 == 0);
+    }
+
+    /**
+     * Converts a timestamp to GMT.
+     *
+     * @param int $time Unix timestamp
+     *
+     * @since 3.0.0
+     *
+     * @return int
+     */
+    public static function timestampToGmt($time = null)
+    {
+        $time = $time ?? time();
+        $dateTime = new \DateTime();
+        $dateTime->setTimestamp($time)->modify('+2 hours');
+        return $dateTime->format('d/m/Y H:i:s');
+
+    }
+
+    /**
+     * Converts the timestamp in to ago form.
+     *
+     * @param int|string $time Timestamp or English textual datetime (http://php.net/manual/en/function.strtotime.php)
+     *
+     * @since 3.0.0
+     *
+     * @return mixed
+     */
+    public static function ago($time)
```

Comment:
```php
function time_elapsed_string($datetime, $full = false) {
    $now = new DateTime;
    $ago = new DateTime($datetime);
    $diff = $now->diff($ago);

    $diff->w = floor($diff->d / 7);
    $diff->d -= $diff->w * 7;

    $string = array(
        'y' => 'year',
        'm' => 'month',
        'w' => 'week',
        'd' => 'day',
        'h' => 'hour',
        'i' => 'minute',
        's' => 'second',
    );
    foreach ($string as $k => &$v) {
        if ($diff->$k) {
            $v = $diff->$k . ' ' . $v . ($diff->$k > 1 ? 's' : '');
        } else {
            unset($string[$k]);
        }
    }

    if (!$full) $string = array_slice($string, 0, 1);
    return $string ? implode(', ', $string) . ' ago' : 'just now';
}
```

To keep the time `ago` simple/lightweight, consider this function.

This approach is about using the [DateTime::diff](https://www.php.net/manual/en/datetime.diff.php) to get the  difference time between two date time.

We can also use the `DateTime` OO style or `date_create` to complete the difference work :).

Reply:
i see but i think it will not work with timestamp

Reply:
i see it will be work if we consider following code lets see
```php
    /**
     * Converts a timestamp to DateTime.
     *
     * @param int|null $time Unix timestamp.
     *
     * @since 3.0.0
     *
     * @return string
     */
    private static function timestampToDate($time = null)
    {
        $time = $time ?? time();
        $time = self::formatTime($time);
        $dateTime = new \DateTime();
        $dateTime->setTimestamp($time);

        return $dateTime->format('d-m-Y H:i:s');
    }

    /**
     * Converts the timestamp in to ago form.
     *
     * @param int|string $time Timestamp or English textual datetime (http://php.net/manual/en/function.strtotime.php)
     *
     * @author https://github.com/peter279k (https://github.com/zestframework/Zest_Framework/pull/206).
     * @since 3.0.0
     *
     * @return mixed
     */
    public static function ago($dateTime, $full = false)
    {
        $dateTime = self::timestampToDate($dateTime);
        $now = new \DateTime;
        $ago = new \DateTime($dateTime);
        $diff = $now->diff($ago);

        $diff->w = floor($diff->d / 7);
        $diff->d -= $diff->w * 7;

        $string = array(
            'y' => ':year',
            'm' => ':month',
            'w' => ':week',
            'd' => ':day',
            'h' => ':hour',
            'i' => ':minute',
            's' => ':second',
        );
        foreach ($string as $k => &$v) {
            if ($diff->$k) {
                $v = $diff->$k . ' ' . $v . ($diff->$k > 1 ? 's' : '');
            } else {
                unset($string[$k]);
            }
        }

        if (!$full) $string = array_slice($string, 0, 1);

        return $string ? implode(', ', $string) . ' ' : ':just';
    }
```

Reply:
FormatTime method
```php
        $time = preg_match('~\D~', $time) ? strtotime($time) : $time;
        return $time;
```

Reply:
It looks good to me at this moment (without unit test).

Reply:
okay i am going to implement later we can fix iif any issue